Tools/cases_generator/tier1_generator.py

```diff
@@ -0,0 +1,417 @@
+"""Generate the main interpreter switch.
+Reads the instruction definitions from bytecodes.c.
+Writes the cases to generated_cases.c.h, which is #included in ceval.c.
+"""
+
+import argparse
+import os.path
+import sys
+
+from analyzer import (
+    Analysis,
+    Instruction,
+    Uop,
+    Part,
+    analyze_files,
+    Skip,
+    StackItem,
+    analysis_error,
+)
+from cwriter import CWriter
+from typing import TextIO, Iterator
+from lexer import Token
+from stack import StackOffset
+
+
+HERE = os.path.dirname(__file__)
+ROOT = os.path.join(HERE, "../..")
+THIS = os.path.relpath(__file__, ROOT).replace(os.path.sep, "/")
+
+DEFAULT_INPUT = os.path.relpath(os.path.join(ROOT, "Python/bytecodes.c"))
+DEFAULT_OUTPUT = os.path.relpath(os.path.join(ROOT, "Python/generated_cases.c.h"))
```

Comment:
pathlib might make this slightly more elegant:

```py
from pathlib import Path

HERE = Path(__file__).parent
ROOT = HERE.parent.parent
THIS = Path(__file__).relative_to(ROOT).as_posix()

DEFAULT_INPUT = (ROOT / "Python/bytecodes.c").absolute()
DEFAULT_OUTPUT = (ROOT / "Python/generated_cases.c.h").absolute()
```

Reply:
That is better, but I think there are enough changes in this PR for now.
I'll need to refactor this code to split off the other generators, so I can use pathlib then.